src/test/groovy/org/prebid/server/functional/tests/ImpRequestSpec.groovy

```diff
@@ -0,0 +1,203 @@
+        def extPmp = Pmp.defaultPmp
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            imp.first.tap {
+                pmp = Pmp.defaultPmp
+                ext.prebid.imp = [(aliasName): new Imp(pmp: extPmp)]
+            }
+            ext.prebid.aliases = [(aliasName.value): GENERIC]
+        }
+
+        when: "Requesting PBS auction"
+        defaultPbsServiceWithAlias.sendAuctionRequest(bidRequest)
+
+        then: "BidderRequest should update imp information based on imp.ext.prebid.imp value"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.imp.pmp == [extPmp]
+
+        and: "PBS should remove imp.ext.prebid.imp from bidderRequest"
+        assert !bidderRequest?.imp?.ext?.prebid?.imp
+
+        and: "PBS should remove imp.ext.prebid.bidder from bidderRequest"
+        assert !bidderRequest?.imp?.first?.ext?.prebid?.bidder
+
+        where:
+        aliasName        | bidderName
+        ALIAS            | GENERIC
+        ALIAS_CAMEL_CASE | GENERIC
+        ALIAS            | GENERIC_CAMEL_CASE
+        ALIAS_CAMEL_CASE | GENERIC_CAMEL_CASE
+    }
+
+    def "PBS shouldn't update imp fields when imp.ext.prebid.imp contain only bidder with invalid name"() {
+        given: "Default basic BidRequest"
+        def impPmp = Pmp.defaultPmp
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            imp.first.tap {
+                pmp = impPmp
+                ext.prebid.imp = [(bidderName): new Imp(pmp: Pmp.defaultPmp)]
+            }
+        }
+
+        when: "Requesting PBS auction"
+        def response = defaultPbsServiceWithAlias.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain warning"
+        assert response.ext.warnings[PREBID]?.code == [999]
+        assert response.ext.warnings[PREBID]?.message ==
+                ["WARNING: request.imp[0].ext.prebid.imp.${bidderName} was dropped with the reason: invalid bidder"]
+
+        and: "BidderRequest shouldn't update imp information based on imp.ext.prebid.imp value"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.imp.pmp == [impPmp]
+
+        and: "PBS should remove imp.ext.prebid.imp from bidderRequest"
+        assert !bidderRequest?.imp?.first?.ext?.prebid?.imp
+
+        and: "PBS should remove imp.ext.prebid.bidder from bidderRequest"
+        assert !bidderRequest?.imp?.first?.ext?.prebid?.bidder
+
+        where:
+        bidderName << [WILDCARD, UNKNOWN]
+    }
+
+    def "PBS should validate imp and add proper warning when imp.ext.prebid.imp contain invalid ortb data"() {
+        given: "BidRequest with invalid config for ext.prebid.imp"
+        def impPmp = Pmp.defaultPmp
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            imp.first.tap {
+                pmp = impPmp
+                ext.prebid.imp = [(GENERIC): Imp.defaultImpression.tap {
+                    id = ""
```

Comment:
Maybe better to add const like EMPTY_ID and also assign null value in where block